### PR TITLE
[lldb][prototype] Unwinding swift async functions

### PR DESCRIPTION
**Pull request by felipepiovezan**

### CODE REVIEW

File: lldb/source/Core/FormatEntity.cpp [hunk 1]
```diff
@@ -437,6 +440,15 @@ static bool DumpAddressAndContent(Stream &s, const SymbolContext *sc,
   return false;
 }
 
+lldb::addr_t GetAsyncContext(RegisterContext *regctx) {
+  if (!regctx)
+    return LLDB_INVALID_ADDRESS;
+  auto arch = regctx->CalculateTarget()->GetArchitecture();
+  auto reg = regctx->ConvertRegisterKindToRegisterNumber(
+      RegisterKind::eRegisterKindDWARF, arm64_dwarf::x22);
```
Comment (jasonmolenda) [inline lldb/source/Core/FormatEntity.cpp:448]: x86_64 is needed too (I see you got `arch` here just before this, I expect you intend/intended to use it)

File: lldb/source/Core/FormatEntity.cpp [hunk 2]
```diff
@@ -1796,6 +1808,36 @@ bool FormatEntity::Format(const Entry &entry, Stream &s,
...
+      auto read_addr = [&](addr_t addr) {
+        auto *process = exe_ctx->GetProcessPtr();
+        lldb::addr_t value_read = LLDB_INVALID_ADDRESS;
+        Status error;
+        process->ReadMemory(addr, &value_read, 8, error);
+        return value_read;
+      };
+
+      if (auto *frame = exe_ctx->GetFramePtr()) {
+        s.Printf("x22 = 0x%" PRIx64, GetAsyncContext(frame->GetRegisterContext().get()));
```
Comment (jasonmolenda) [inline lldb/source/Core/FormatEntity.cpp:1822]: also handle intel reg name.

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp [hunk 3]
```diff
@@ -2582,37 +2584,92 @@ lldb::addr_t SwiftLanguageRuntime::GetAsyncContext(RegisterContext *regctx) {
...
+  UnwindPlanSP unwind_plan =
+      func_unwinders->GetUnwindPlanAtNonCallSite(target, regctx.GetThread());
+  if (!unwind_plan)
+    return {};
+
+  UnwindPlan::RowSP row = unwind_plan->GetRowForFunctionOffset(
+      pc.GetFileAddress() - func_start_addr.GetFileAddress());
+  UnwindPlan::Row::AbstractRegisterLocation regloc;
+  if (!row->GetRegisterInfo(regnums.async_ctx_regnum, regloc)) {
+    // If the plan doesn't have register about the register, we can use its
```
Comment (jasonmolenda) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp]: "If the plan doesn't have an entry for the register, it is unmodified from it's caller value"

**Comment by felipepiovezan** [comment]
Closing in favor of the real PR: https://github.com/swiftlang/llvm-project/pull/9386

**Pull request closed without merging**